pkg/vmprovider/providers/vsphere/virtualmachine/backup.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package virtualmachine
+
+import (
+	goctx "context"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	corev1 "k8s.io/api/core/v1"
+	ctrlruntime "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha1"
+	"github.com/vmware-tanzu/vm-operator/pkg/context"
+	"github.com/vmware-tanzu/vm-operator/pkg/util"
+	"github.com/vmware-tanzu/vm-operator/pkg/vmprovider/providers/vsphere/constants"
+)
+
+// BackupVirtualMachine backs up the VM's kube data and bootstrap data into the VM's ExtraConfig.
```

Comment:
Expand the comment to indicate what kube data means here.  Info like why we don't care about persisting the status, or why do we not wipe out the managed fields will be helpful here.